tests/test_json_excel_compare.py

```diff
@@ -0,0 +1,146 @@
+        self.json_topics = xml.hvac_topics.copy()
+
+    async def test_compare_csv_and_json(self):
+        self.load_csv()
+        self.load_json()
+        json_topics_missing_in_csv = []
+        csv_topics_missing_in_json = []
+        topic_types_differing = []
+        idl_types_differing = []
+        units_differing = []
+        limits_differing = []
+
+        for json_topic in sorted(self.json_topics.keys()):
+            if json_topic not in sorted(self.csv_topics.keys()):
+                json_topics_missing_in_csv.append(json_topic)
+
+        for csv_topic in sorted(self.csv_topics.keys()):
+            if csv_topic not in sorted(self.json_topics.keys()):
+                csv_topics_missing_in_json.append(csv_topic)
+
+        for json_topic in sorted(self.json_topics.keys()):
+            if json_topic in sorted(self.csv_topics.keys()):
+                if (
+                    self.json_topics[json_topic]["topic_type"]
+                    != self.csv_topics[json_topic]["topic_type"]
+                ):
+                    topic_types_differing.append(
+                        (
+                            json_topic,
+                            self.json_topics[json_topic]["topic_type"],
+                            self.csv_topics[json_topic]["topic_type"],
+                        )
+                    )
+                if (
+                    self.json_topics[json_topic]["idl_type"]
+                    != self.csv_topics[json_topic]["idl_type"]
+                ):
+                    idl_types_differing.append(
+                        (
+                            json_topic,
+                            self.json_topics[json_topic]["idl_type"],
+                            self.csv_topics[json_topic]["idl_type"],
+                        )
+                    )
+                if (
+                    self.json_topics[json_topic]["unit"]
+                    != self.csv_topics[json_topic]["unit"]
+                ):
+                    units_differing.append(
+                        (
+                            json_topic,
+                            self.json_topics[json_topic]["unit"],
+                            self.csv_topics[json_topic]["unit"],
+                        )
+                    )
+                if (
+                    self.json_topics[json_topic]["limits"]
+                    != self.csv_topics[json_topic]["limits"]
+                ):
+                    limits_differing.append(
+                        (
+                            json_topic,
+                            self.json_topics[json_topic]["limits"],
+                            self.csv_topics[json_topic]["limits"],
+                        )
+                    )
+
+        if (
+            json_topics_missing_in_csv
+            or csv_topics_missing_in_json
+            or topic_types_differing
+            or idl_types_differing
+            or units_differing
+            or limits_differing
+        ):
```

Comment:
You could skip big "if" statement with the following (or if you dislike the output from fail, you can print the error messages first or log them in a loop. But it looks OK to me).
```
err_msgs = []
if json_topics_missing_in_csv:
    err_msgs.append( f"JSON topics missing in CSV: {sorted(json_topics_missing_in_csv)}")
...
if err_msgs:
    self.fail("Differences found:\n" + "\n".join(err_msgs))
```